src/main/java/org/jenkinsci/plugins/credentialsbinding/masking/Base64SecretPatternFactory.java

```diff
@@ -0,0 +1,60 @@
+package org.jenkinsci.plugins.credentialsbinding.masking;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.Collections;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class Base64SecretPatternFactory implements SecretPatternFactory {
+    @NonNull
+    @Override
+    public Collection<String> getEncodedForms(@NonNull String input) {
+        return getBase64Forms(input);
+    }
+
+    @NonNull
+    private Collection<String> getBase64Forms(@NonNull String secret) {
+        if (secret.length() == 0) {
+            return Collections.emptyList();
+        }
+
+        Base64.Encoder encoder = Base64.getEncoder();
```

Comment:
Maybe we should cover variants from `getUrlEncoder` as well? That would handle `+/` → `-_` which might be used on occasion.

(`getMimeEncoder` seems pointless, since if there are newlines in the Base64 form then line-by-line masking is not going to work anyway.)

Reply:
Good point, addressed by [d685136](https://github.com/jenkinsci/credentials-binding-plugin/pull/269/commits/d68513672d0287dca7bf97cbd62e1f89daa8eef5).